config/initializers/sidekiq.rb

```diff
@@ -0,0 +1,20 @@
+require 'sidekiq'
+require 'sidekiq-cron'
+require 'sidekiq/web'
+require 'active_support/security_utils'
+
+schedule_file = "config/scheduled_tasks.yml"
+
+if File.exist?(schedule_file) && Sidekiq.server?
+  Sidekiq::Cron::Job.load_from_hash YAML.load_file(schedule_file)
+end 
+
+Sidekiq::Web.use(Rack::Auth::Basic) do |user, password|
+  # Protect against timing attacks:
+  # - See https://codahale.com/a-lesson-in-timing-attacks/
+  # - See https://thisdata.com/blog/timing-attacks-against-string-comparison/
+  # - Use & (do not use &&) so that it doesn't short circuit.
+  # - Use digests to stop length information leaking
+  ActiveSupport::SecurityUtils.secure_compare(user, ENV["SIDEKIQ_ADMIN_USER"]) &
```

Comment:
We may want to pull the admin user and password out into variables and then do something like we do for the oauth credentials in civic 

```
password = ENV['SIDEKIQ_ADMIN_USER'] || Rails.secrets.sidekiq_admin_user

```

(or whatever the exact syntax is) so that we can use env variables locally and the capistrano deployed secrets file in prod.